client/shared/src/search/parser/parser.ts

```diff
@@ -272,6 +272,122 @@ const scanToken = <T extends Term = Literal>(
     }
 }
 
+const keepScanning = (input: string, start: number): boolean => {
+    const result = oneOf<Literal | Sequence>(filterKeyword, followedBy(operator, whitespace))(input, start)
+    return result.type !== 'success'
+}
```

Comment:
```suggestion
const scanFilterOrOperator = oneOf<Literal | Sequence>(filterKeyword, followedBy(operator, whitespace))

const keepScanning = (input: string, start: number): boolean => scanFilterOrOperator(input, start).type !== 'success'
```

Extracting the parser declaration in an outer scope would avoid doing extra work every time `keepScanning` is called. I also think it would improve readability.

Reply:
![](https://media2.giphy.com/media/1hAxQTH0HEWS3L0oRF/200.gif?cid=5a38a5a2qzul6ur5eks9j5vs299swpbsepsn2bftl7uldmeg&rid=200.gif)

Nice, yeah I'll get better at the scrunching in TS